app/auth-portal/src/content/tutorial/08-asset_authoring.md

```diff
@@ -0,0 +1,442 @@
+                .setKind("text")
+                .build())
+        .build();
+
+    const sizeProp = new PropBuilder()
+        .setKind("integer")
+        .setName("size")
+        .setWidget(
+            new PropWidgetDefinitionBuilder()
+                .setKind("text")
+                .build())
+        .build();
+
+    const volumeTypeProp = new PropBuilder()
+        .setKind("string")
+        .setName("volumeType")
+        .setDefaultValue("gp2")
+        .setWidget(
+            new PropWidgetDefinitionBuilder()
+                .setKind("comboBox")
+                .addOption("Magnetic (standard)", "standard")
+                .addOption("Provisioned IOPS SSD (io1)", "io1")
+                .addOption("Provisioned IOPS SSD (io2)", "io2")
+                .addOption("General Purpose SSD (gp2)", "gp2")
+                .addOption("General Purpose SSD (gp3)", "gp3")
+                .addOption("Cold HDD (sc1)", "sc1")
+                .addOption("Throughput Optimized HDD (st1)", "st1")
+                .build())
+        .build();
+
+    const regionSocket = new SocketDefinitionBuilder()
+        .setArity("one")
+        .setName("Region")
+        .build();
+
+    const volumeIdSocket = new SocketDefinitionBuilder()
+        .setArity("one")
+        .setName("Volume ID")
+        .build();
+
+    return new AssetBuilder()
+        .addProp(availabilityZoneProp)
+        .addProp(regionProp)
+        .addProp(iopsProp)
+        .addProp(sizeProp)
+        .addProp(volumeTypeProp)
+        .addInputSocket(regionSocket)
+        .addOutputSocket(volumeIdSocket)
+        .build()
+}
+```
+
+From this code, we can see that a prop uses a `PropBuilder` class and needs a `kind` and a
+`name`. It also needs a `widget`. A widget is a way for us to be able to interact with the property via the System
+Initiative modelling screen.
+
+You can notice that within the widgets we have combo boxes to help our colleagues.
+We have also added a default value for `volumeType`. A volume type is a requirement when choosing
+an EBS Volume. AWS defaults to `gp2` in the UI so we can follow suit here in our schema using `setDefaultValue` in the prop.
+
+A prop can get its value from another `prop` or `inputSocket`. We can use the `setValueFrom` class as part of a prop to be
+able to do this. We can see an example of this in the `regionProp` in our schema definition.
+
+```js
+const regionProp = new PropBuilder()
+    .setKind("string")
+    .setName("region")
+    .setWidget(new PropWidgetDefinitionBuilder().setKind("text").build())
+    .setValueFrom(new ValueFromBuilder()
+        .setKind("inputSocket")
+        .setSocketName("Region")
+        .build())
+    .build();
+```
+
+The last parts of the schema definition are the input and output sockets. In System Initiative, we
+can model our infrastructure using edges to other assets. We currently support connecting resources
+to a `region` asset so we can create an input socket for a Region.
+
```

Comment:
[perhaps clarifying how frames relate to input and output sockets might be useful? Suggestion below] 

The last parts of the schema definition are the input and output sockets. In System Initiative, we
can model our infrastructure using edges to other assets, including 
-  a connection between an output socket and an input socket
- a connection between a configuration frame (where all the assets inside the frame are configured by it) and an input socket. 

You can see in the schema definition we have added a region input socket, so that it can be configured by a Region frame.